### DISCUSSION TITLE
How to disable external-host-error?

### DISCUSSION THREAD

**Discussion by dudicoco** [renovatebot/renovate#12968] [Request Help]
Hi,

I have enabled the [helm-values](https://docs.renovatebot.com/modules/manager/helm-values) manager.

We are using the GitHub hosted renovate app.

Within our helm charts we have images from a private docker registry, renovate fails on these with `external-host-error`:

```json
{
  "registryHost": "https://index.docker.io",
  "dockerRepository": "my-registry/amazing-app",
  "err": {
    "name": "HTTPError",
    "code": "ERR_NON_2XX_3XX_RESPONSE",
    "timings": {
      "start": 1638785857823,
      "socket": 1638785857823,
      "lookup": 1638785857823,
      "connect": 1638785857895,
      "secureConnect": 1638785858041,
      "upload": 1638785858041,
      "response": 1638785858120,
      "end": 1638785858120,
      "phases": {
        "wait": 0,
        "dns": 0,
        "tcp": 72,
        "tls": 146,
        "request": 0,
        "firstByte": 79,
        "download": 0,
        "total": 297
      }
    },
    "message": "Response code 401 (Unauthorized)",
    "stack": "HTTPError: Response code 401 (Unauthorized)\n    at Request.<anonymous> (/home/ubuntu/renovateapp/node_modules/got/dist/source/as-promise/index.js:117:42)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)",
    "options": {
      "headers": {
        "user-agent": "Renovate Bot (GitHub App 2740)",
        "authorization": "***********",
        "accept": "application/json",
        "accept-encoding": "gzip, deflate, br"
      },
      "url": "https://index.docker.io/v2/my-registry/amazing-app/tags/list?n=10000",
      "hostType": "docker",
      "username": "",
      "password": "",
      "method": "GET",
      "http2": false
    },
    "response": {
      "statusCode": 401,
      "statusMessage": "Unauthorized",
      "body": {
        "errors": [
          {
            "code": "UNAUTHORIZED",
            "message": "authentication required",
            "detail": [
              {
                "Type": "repository",
                "Class": "",
                "Name": "my-registry/amazing-app",
                "Action": "pull"
              }
            ]
          }
        ]
      },
      "headers": {
        "content-type": "application/json",
        "docker-distribution-api-version": "registry/2.0",
        "www-authenticate": "Bearer realm=\"https://auth.docker.io/token\",service=\"registry.docker.io\",scope=\"repository:my-registry/amazing-app:pull\",error=\"insufficient_scope\"",
        "date": "Mon, 06 Dec 2021 10:17:38 GMT",
        "content-length": "166",
        "strict-transport-security": "max-age=31536000",
        "connection": "close"
      },
      "httpVersion": "1.1"
    }
  }
}
```

First, I find it very strange that the default behaviour is for renovate to fail the run, unless you go to the renovate logs you have no way of knowing that renovate is failing and has stopped working, which is very problematic. I'm not sure if there's an existing issue for this problem.

I have tried disabling the errors using the following configs:
```json
{
  "extends": [
    "config:base"
  ],
  "helm-values": {
    "packageRules": [
      {
        "excludePackagePrefixes": [
          "my-registry"
        ]
      }
    ]
  }
}
```

```json
{
  "extends": [
    "config:base"
  ],
  "helm-values": {
    "hostRules": [
      {
        "matchHost": "my-registry",
        "enabled": false
      }
    ]
  }
}
```

```json
{
  "extends": [
    "config:base"
  ],
  "helm-values": {},
  "hostRules": [
    {
      "abortOnError": false
    }
  ]
}
```

None of these configs helped.

What is the correct way to solve this problem?

Thanks

**Comment by rarkins** [comment]
There are multiple ways to do it, but this is probably most easy to understand:

```
{
  "extends": [
    "config:base"
  ],
    "packageRules": [
      {
        "matchDataSources": ["docker"],
        "matchPackagePrefixes": [
          "my-registry/"
        ],
        "enabled": false
      }
    ]
}
```

**Reply by dudicoco** [reply]
Thanks @rarkins, that worked.

Needed to use `matchDatasources` instead of `matchDataSources`.


Is there a way to completely disable renovate failing a run on any error? Similar to the this config which i've tried (which didn't work):
```json
{
  "extends": [
    "config:base"
  ],
  "hostRules": [
    {
      "abortOnError": false
    }
  ]
}
```

**Reply by rarkins** [reply]
There might be a way to get that option to work, but of what you tried doesn't then it probably needs some debugging to work out why. Be warned that temporary outages of registries would result in PRs getting autoclosed though.

**Reply by dudicoco** [reply]
@rarkins what kind of debugging is required here? Isn't it a pretty straight forward config option?
I am fine with PRs getting autoclosed versus the alternative which is renovate failing silently without me knowing about it. Per my first post in the discussion I think that this default behaviour should be changed.

**Reply by rarkins** [reply]
You can try debugging it and find out. Check our docs/ folder

**Reply by viceice** [reply]
Maybe he needs https://docs.renovatebot.com/configuration-options/#abortignorestatuscodes

**Reply by dudicoco** [reply]
> You can try debugging it and find out. Check our docs/ folder

The only doc I've found that seems relevant is https://github.com/renovatebot/renovate/blob/main/docs/usage/troubleshooting.md which just instructs to check the logs for the GH app, which i've already did.

**Reply by dudicoco** [reply]
> Maybe he needs https://docs.renovatebot.com/configuration-options/#abortignorestatuscodes

Thanks, though i'm looking for a way to disable all errors from renovate (do not fail runs due to errors).

**Reply by rarkins** [reply]
> > You can try debugging it and find out. Check our docs/ folder
> 
> The only doc I've found that seems relevant is [`main`/docs/usage/troubleshooting.md](https://github.com/renovatebot/renovate/blob/main/docs/usage/troubleshooting.md) which just instructs to check the logs for the GH app, which i've already did.

https://github.com/renovatebot/renovate/blob/cf95251dee3f26c64f003e3caf31ea85dc931ba6/docs/development/local-development.md

**Reply by dudicoco** [reply]
@rarkins i'm not sure I follow, you suggest that I build and run renovate locally in order to debug this issue?

**Reply by rarkins** [reply]
@dudicoco the only way to know why the field isn't having an effect is to debug it locally. If you care about it and have the time/capabilities to do your own debugging, then I've pointed you to how. But if not, then you'll need to use the provided workaround instead.
